addons/stock/models/stock_lot.py

```diff
@@ -149,12 +151,15 @@ def _compute_delivery_ids(self):
     def _compute_last_delivery_partner_id(self):
         serial_products = self.filtered(lambda l: l.product_id.tracking == 'serial')
         delivery_ids_by_lot = serial_products._find_delivery_ids_by_lot()
-        (self - serial_products).last_delivery_partner_id = False
+        (self - serial_products).last_delivery_partner_id = (self - serial_products).last_delivery_date = (self - serial_products).quantity = False
         for lot in serial_products:
             if lot.product_id.tracking == 'serial' and len(delivery_ids_by_lot[lot.id]) > 0:
-                lot.last_delivery_partner_id = self.env['stock.picking'].browse(delivery_ids_by_lot[lot.id]).sorted(key='date_done', reverse=True)[0].partner_id
+                recent_delivery = self.env['stock.picking'].browse(delivery_ids_by_lot[lot.id]).sorted(key='date_done', reverse=True)[0]
+                lot.last_delivery_partner_id = recent_delivery.partner_id
+                lot.last_delivery_date = recent_delivery.date_done
+                lot.quantity = recent_delivery.move_ids.move_line_ids.mapped("quantity")[0]
```

Comment:
I know qty will always be 1 but get it dynamic by mapping lot name with delivery

Reply:
Done.